src/shared/xen-util.h

```diff
@@ -0,0 +1,20 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#if HAVE_XENCTRL
+#  include <xenctrl.h>
+#endif
+
+#if HAVE_XENCTRL
+int dlopen_xenctrl(void);
+#else
+static inline int dlopen_xenctrl(void) {
+        return -EOPNOTSUPP;
+}
+#endif
```

Comment:
Please implement this as we do for other dlopen() functions (put the ifdef in the function implementation instead of in the header)